### ISSUE TITLE
src/URI/ByteString/Internal.hs:172:47: Could not deduce (Data.String.IsString Builder)

### ISSUE THREAD

**Issue by juhp** [Soostone/uri-bytestring#37]
uri-bytestring-0.2.3.1 failed to build for me in a cabal sandbox.
I am not sure why (local cabal install worked okay).

```
src/URI/ByteString/Internal.hs:172:47:
    Could not deduce (Data.String.IsString Builder)
      arising from the literal ‘"/"’
    from the context (Relative ~ Relative)
      bound by a pattern with constructor
                 RelativeRef :: Maybe Authority
                                -> ByteString -> Query -> Maybe ByteString -> URIRef Relative,
               in an equation for ‘normalizeRelativeRef’
      at src/URI/ByteString/Internal.hs:168:61-76
    In the expression: "/"
    In an equation for ‘path’:
        path
          | unoSlashEmptyPath && BS8.null rrPath = "/"
          | otherwise
          = mconcat (intersperse (c8 '/') (map urlEncodePath segs))
    In an equation for ‘normalizeRelativeRef’:
        normalizeRelativeRef
          o@(URINormalizationOptions {..})
          mScheme
          (RelativeRef {..})
          = authority <> path <> query <> fragment
          where
              path
                | unoSlashEmptyPath && BS8.null rrPath = "/"
                | otherwise
                = mconcat (intersperse (c8 '/') (map urlEncodePath segs))
              segs = dropSegs (BS.split slash (pathRewrite rrPath))
              pathRewrite
                | unoRemoveDotSegments = removeDotSegments
                | otherwise = id
              dropSegs [] = []
              dropSegs (h : t)
                | unoDropExtraSlashes = h : (filter (not . BS8.null) t)
                | otherwise = h : t
              ....
cabal: Leaving directory '/home/petersen/tmp/cabal-tmp-32181/uri-bytestring-0.2.3.1'
Failed to install uri-bytestring-0.2.3.1
```

**Comment by MichaelXavier** [comment]
Hi, could you try the `isstring-builder` branch and let me know if that works for you?

**Comment by juhp** [comment]
Sorry for not commenting earlier: it is hard for me to test since it occurs when I am building other packages with cabal...

But I still see this for example when building stack-1.5.1 with ghc-7.10.3:

https://copr-be.cloud.fedoraproject.org/results/petersen/stack/fedora-25-x86_64/00684662-stack/build.log.gz

```
Preprocessing library uri-bytestring-0.3.0.1...
[1 of 5] Compiling URI.ByteString.Types ( src/URI/ByteString/Types.hs, dist/dist-sandbox-34529f35/build/URI/ByteString/Types.o )
[2 of 5] Compiling URI.ByteString.Lens ( src/URI/ByteString/Lens.hs, dist/dist-sandbox-34529f35/build/URI/ByteString/Lens.o )
[3 of 5] Compiling URI.ByteString.Internal ( src/URI/ByteString/Internal.hs, dist/dist-sandbox-34529f35/build/URI/ByteString/Internal.o )
src/URI/ByteString/Internal.hs:174:48:
    Could not deduce (Data.String.IsString Builder)
      arising from the literal `"/"'
    from the context (Relative ~ Relative)
      bound by a pattern with constructor
                 RelativeRef :: Maybe Authority
                                -> ByteString -> Query -> Maybe ByteString -> URIRef Relative,
               in an equation for `normalizeRelativeRef'
      at src/URI/ByteString/Internal.hs:170:61-76
    In the expression: "/"
    In an equation for `path':
        path
          | unoSlashEmptyPath && BS8.null rrPath = "/"
          | segs == [""] = "/"
          | otherwise
          = mconcat (intersperse (c8 '/') (map urlEncodePath segs))
    In an equation for `normalizeRelativeRef':
        normalizeRelativeRef
          o@(URINormalizationOptions {..})
          mScheme
          (RelativeRef {..})
          = authority <> path <> query <> fragment
          where
              path
                | unoSlashEmptyPath && BS8.null rrPath = "/"
                | segs == [""] = "/"
                | otherwise
                = mconcat (intersperse (c8 '/') (map urlEncodePath segs))
              segs = dropSegs (BS.split slash (pathRewrite rrPath))
              pathRewrite
                | unoRemoveDotSegments = removeDotSegments
                | otherwise = id
              dropSegs [] = []
              dropSegs (h : t)
                | unoDropExtraSlashes = h : (filter (not . BS8.null) t)
                | otherwise = h : t
              ....
```

**Comment by MichaelXavier** [comment]
That's not surprising, the branch I made was never released. I'll just merge that branch anyways since I guess its a bit more explicit.

**Comment by MichaelXavier** [comment]
Released at 0.3.0.2
